core/ecschema-editing/src/Differencing/SchemaDifferenceValidator.ts

```diff
@@ -0,0 +1,502 @@
+   * @internal
+   */
+  public async visitKindOfQuantityDifference(entry: KindOfQuantityDifference) {
+    const kindOfQuantity = await this._targetSchema.getItem<KindOfQuantity>(entry.itemName);
+    if (!await this.visitSchemaItemDifference(entry, kindOfQuantity)) {
+      return;
+    }
+
+    if (entry.changeType === "modify" && kindOfQuantity !== undefined) {
+      if (entry.difference.persistenceUnit) {
+        this.addConflict({
+          code: ConflictCode.ConflictingPersistenceUnit,
+          schemaType: SchemaItemType.KindOfQuantity,
+          itemName: kindOfQuantity.name,
+          source: entry.difference.persistenceUnit,
+          target: resolveLazyItemName(kindOfQuantity.persistenceUnit),
+          description: "Kind of Quantity has a different persistence unit.",
+        });
+      }
+    }
+  }
+
+  /**
+   * Visitor implementation for handling MixinClassDifference.
+   * @internal
+   */
+  public async visitMixinDifference(entry: MixinClassDifference) {
+    await this.visitClassDifference(entry);
+  }
+
+  /**
+   * Visitor implementation for handling PhenomenonDifference.
+   * @internal
+   */
+  public async visitPhenomenonDifference(entry: PhenomenonDifference) {
+    await this.visitSchemaItemDifference(entry, await this._targetSchema.getItem(entry.itemName));
+  }
+
+  /**
+   * Visitor implementation for handling ClassPropertyDifference.
+   * @internal
+   */
+  public async visitPropertyDifference(entry: ClassPropertyDifference) {
+    const targetClass = await this.getSchemaItem<ECClass>(entry.itemName);
+    const targetProperty = await targetClass.getProperty(entry.path);
+
+    if (entry.changeType === "add" && targetProperty !== undefined) {
+      const sourceClass = await this._sourceSchema.getItem(targetProperty.class.name) as ECClass;
+      const sourceProperty = await sourceClass.getProperty(targetProperty.name) as Property;
+
+      this.addConflict({
+        code: ConflictCode.ConflictingPropertyName,
+        schemaType: targetProperty.class.schemaItemType,
+        itemName: targetProperty.class.name,
+        path: targetProperty.name,
+        source: resolvePropertyTypeName(sourceProperty),
+        target: resolvePropertyTypeName(targetProperty),
+        description: "Target class already contains a property with a different type.",
+      });
+
+      return;
+    }
+
+    if (entry.changeType === "modify" && targetProperty !== undefined) {
+      if (entry.difference.kindOfQuantity) {
+        this.addConflict({
+          code: ConflictCode.ConflictingPropertyKindOfQuantity,
+          schemaType: targetClass.schemaItemType,
+          itemName: targetClass.name,
+          path: targetProperty.name,
+          source: entry.difference.kindOfQuantity,
+          target: resolveLazyItemName(targetProperty.kindOfQuantity),
+          description: "The property has different kind of quantities defined.",
+        });
+      }
```

Comment:
There is nothing wrong with changing the koq so long as both KOQs have the same persistence unit.  Changing the persistence Unit when you change KOQs is not allowed ... everything else about the KOQs can be different